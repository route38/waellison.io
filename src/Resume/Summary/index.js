import "./index.css";
import React from "react";

class Summary extends React.Component {
    constructor(props) {
        super(props);
    }
    
    render() {
        return (
            <div id="summary">
                <h1>Summary</h1>
                <p>{this.props.text}</p>
            </div>
        );
    }
}

export default Summary;
