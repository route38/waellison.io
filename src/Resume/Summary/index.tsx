import React from "react";

interface IProps {
    text: string;
}

interface IState {

}

class Summary extends React.Component<IProps, IState> {
  render() {
    return (
      <div id="summary">
        <h1>William A. Ellison</h1>
        <h2>Summary</h2>
        <p dangerouslySetInnerHTML={{ __html: this.props.text }} />
      </div>
    );
  }
}

export default Summary;
