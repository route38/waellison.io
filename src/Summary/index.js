import "./index.css";
import React from "react";

class Summary extends React.Component {
    componentDidMount() {
        fetch("./resume.json")
            .then(response => response.json())
            .then(json => this.setState({resume: json}))
            .catch(console.error);
    }

    render() {
        if(this.state) {
            let summary = this.state.resume.summary;
            return (
                <div id="summary">
                    <h1>Summary</h1>
                    <p>{summary}</p>
                </div>
            );
        }
        return null;
    }
}

export default Summary;
