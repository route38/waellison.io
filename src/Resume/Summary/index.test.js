import React from "react";
import renderer from "react-test-renderer";
import Summary from './index';

it('renders correct summary component', () => {
    const component = renderer.create(<Summary text="This is the summary." />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
