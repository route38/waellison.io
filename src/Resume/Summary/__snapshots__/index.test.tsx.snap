// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct summary component 1`] = `
<div
  id="summary"
>
  <h1>
    William A. Ellison
  </h1>
  <h2>
    Summary
  </h2>
  <p
    dangerouslySetInnerHTML={
      Object {
        "__html": "This is the summary.",
      }
    }
  />
</div>
`;
